### PR TITLE
chore: add license check/generation

### PR DESCRIPTION
**Pull request by bernot-dev**
Add `addlicense` checks to all source files, consistent with Google standards.

The `make regen` target is run in check mode in CI, so this will be enforced on new PRs.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -137,15 +137,17 @@ bin: $(GOCMDS) $(GOINSTR)
 regen:       ## Refresh autogenerated files and reformat code.
              ## Use CHECK=1 to only validate clean repo after run.
              ##
-regen:
+regen: $(ADDLICENSE)
```
Comment (TheSpiritXIII) [inline Makefile:140]: question: maybe I'm missing something but where did this come from? I don't see this defined anywhere.

**Review by TheSpiritXIII — APPROVED** [review]
Just added a question. Thanks!

File: Makefile (hunk 1 above)
Comment (bernot-dev) [reply Makefile:140]: It's in Bingo. The dependency was already added with the Helm change. This is just expanding it to check the rest of the repo.

https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/.bingo/Variables.mk#L20-L24

**Review by pintohutch — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (TheSpiritXIII) [reply Makefile:140]: Thanks! Looks good!

**Pull request merged**